lightning-persister/src/lib.rs

```diff
@@ -108,39 +102,64 @@ impl FilesystemPersister {
 		util::write_to_file(path, "manager".to_string(), manager)
 	}
 
-	#[cfg(test)]
-	fn load_channel_data<Keys: KeysInterface>(&self, keys: &Keys) ->
-		Result<HashMap<OutPoint, ChannelMonitor<Keys::Signer>>, ChannelMonitorUpdateErr> {
-			if let Err(_) = fs::create_dir_all(self.path_to_monitor_data()) {
-				return Err(ChannelMonitorUpdateErr::PermanentFailure);
+	/// Read `ChannelMonitor`s from disk.
+	pub fn read_channelmonitors<Signer: Sign, K: Deref> (
+		&self, keys_manager: K
+	) -> Result<HashMap<OutPoint, (BlockHash, ChannelMonitor<Signer>)>, std::io::Error>
+		where K::Target: KeysInterface<Signer=Signer> + Sized
+	{
+		let path = self.path_to_monitor_data();
+		if !Path::new(&path).exists() {
+			return Ok(HashMap::new());
+		}
+		let mut outpoint_to_channelmonitor = HashMap::new();
+		for file_option in fs::read_dir(path).unwrap() {
+			let file = file_option.unwrap();
+			let owned_file_name = file.file_name();
+			let filename = owned_file_name.to_str();
+			if !filename.is_some() || !filename.unwrap().is_ascii() || filename.unwrap().len() < 65 {
+				return Err(std::io::Error::new(
+					std::io::ErrorKind::Other,
```

Comment:
`InvalidData` would be a more suitable error kind here and elsewhere.

Reply:
Hmm, I think the custom error messages are more helpful?

Reply:
Oh, you can keep the custom error message. I'm just saying to replace `std::io::ErrorKind::Other` with `std::io::ErrorKind::InvalidData`.